### PR TITLE
Add spellcheck JS-API

### PR DESCRIPTION
**Pull request by eromano**
**Please check if the PR fulfills these requirements**
```
[ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-js-api/wiki/Commit-format)
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: test/content-services/webScriptApi.spec.ts [hunk 1]
```diff
@@ -73,38 +73,38 @@ describe('WebScript', function () {
...
 
-            let webscriptPormise = this.alfrescoJsApi.core.webscriptApi.executeWebScript('GET', this.scriptPath, null, this.contextRoot, this.servicePath);
+            let webscriptPrormise = this.alfrescoJsApi.core.webscriptApi.executeWebScript('GET', this.scriptPath, null, this.contextRoot, this.servicePath);
 
-            webscriptPormise.catch(() => {
+            webscriptPrormise.catch(() => {
             });
 
-            webscriptPormise.on('error', () => {
+            webscriptPrormise.on('error', () => {
```
Comment (adinapitul) [inline test/content-services/webScriptApi.spec.ts]: misspelling

File: cspell.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    "Fileplans"
+  ],
+  "dictionaries": [
+    "html",
+    "en-gb",
+    "en_US",
+    "softwareTerms"
+  ],
+  "ignorePaths": [
+    "lib/{src/test}/**/*.spec.ts",
```
Comment (adinapitul) [inline cspell.json]: can we let spellcheck enabled for specs?

File: cspell.json [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    "taskactions",
+    "assoctype",
+    "bloggs",
+    "addtional",
+    "identitylinks",
+    "Processinstances",
+    "Processinstancevariables",
+    "Adminendpoints",
+    "jwks",
+    "Prormise",
```
Comment (adinapitul) [inline cspell.json]: should be "Promise"

File: cspell.json [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    "Securitycontrolsettings",
+    "MAXCONTENTSIZEEXCEEDED",
+    "INPROGRESS",
+    "GREATERTHAN",
+    "POJO",
+    "Subprocesses",
+    "Taskvariables",
+    "Taskforms",
+    "Systemproperties",
+    "Froms",
```
Comment (adinapitul) [inline cspell.json]: this one should be "Forms"

File: cspell.json [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+    "Integrationdrive",
+    "Formmodels",
+    "Datasources",
+    "BPMECM",
+    "ECMBPM",
+    "Csrf",
+    "taskactions",
+    "assoctype",
+    "bloggs",
+    "addtional",
```
Comment (adinapitul) [inline cspell.json]: this is a misspelling, "additional"

File: cspell.json [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    "identitylinks",
+    "Processinstances",
+    "Processinstancevariables",
+    "Adminendpoints",
+    "jwks",
+    "Prormise",
+    "respo",
```
Comment (adinapitul) [inline cspell.json]: should be change to "resp"

File: cspell.json [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    "Listner",
```
Comment (adinapitul) [inline cspell.json]: "Listener"

File: cspell.json (hunk 3 above)
Comment (eromano) [reply cspell.json]: done

File: cspell.json (hunk 4 above)
Comment (eromano) [reply cspell.json]: done

File: cspell.json [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+    "Milli",
+    "aswell",
```
Comment (adinapitul) [inline cspell.json]: "as well"

File: cspell.json [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+    "specifc",
```
Comment (adinapitul) [inline cspell.json]: "specific"

File: cspell.json [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+    "Listner",
```
Comment (adinapitul) [inline cspell.json]: duplicated entry

File: cspell.json [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+    "Accountintegration",
+    "auths",
+    "Recordfolders",
+    "utest",
+    "Gssites",
+    "plans's",
```
Comment (adinapitul) [inline cspell.json]: "plan's"

File: cspell.json (hunk 7 above)
Comment (eromano) [reply cspell.json]: Done

File: cspell.json (hunk 11 above)
Comment (eromano) [reply cspell.json]: we need to keep it is in the comments

File: cspell.json [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+    "Idmsync",
+    "Runtimeappdefinitions",
+    "Millis",
+    "NOTEQUALSIGNORECASE",
+    "EQUALSIGNORECASE",
+    "NOTEQUALS",
+    "LESSTHAN",
+    "LESSTHANOREQUALS",
+    "LIKEIGNORECASE",
+    "Represenation",
```
Comment (adinapitul) [inline cspell.json]: misspelling, "Representation"

File: cspell.json [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+    "LIKEIGNORECASE",
+    "basepath",
+    "arbitary",
```
Comment (adinapitul) [inline cspell.json]: "arbitrary"

File: cspell.json [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+    "basepath",
+    "Millis",
+    "Authonfiguration",
+    "Decisiontables",
+    "Runtimeappdeployments",
+    "Scriptfiles",
+    "Userprofile",
+    "stddev",
+    "usewd",
```
Comment (adinapitul) [inline cspell.json]: misspelling, "used"

File: cspell.json (hunk 12 above)
Comment (eromano) [reply cspell.json]: can not change is in the name of the files

File: cspell.json [hunk 15]
```diff
@@ -146,7 +146,7 @@
     "Taskvariables",
     "Taskforms",
     "Systemproperties",
-    "Froms",
+    "Forms",
```
Comment (adinapitul) [inline cspell.json]: I guess no need to exclude this from spellcheck since it's a correct word now

File: src/authentication/oauth2Auth.ts [hunk 16]
```diff
@@ -42,7 +42,7 @@ export class Oauth2Auth extends AlfrescoApiClient {
         'oauth2': { accessToken: '' }, type: 'oauth2', 'basicAuth': {}
     });
 
-    iFameHashListner: any;
+    iFameHashListener: any;
```
Comment (adinapitul) [inline src/authentication/oauth2Auth.ts]: Frame instead of Fame

**Pull request merged**